File: src/test/java/no/nav/fo/veilarbregistrering/arbeidsforhold/adapter/StubAaregRestClient.java
package no.nav.fo.veilarbregistrering.arbeidsforhold.adapter;

import no.nav.common.sts.SystemUserTokenProvider;
import no.nav.fo.veilarbregistrering.FileToJson;
import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer;

import static org.mockito.Mockito.mock;

class StubAaregRestClient extends AaregRestClient {

    StubAaregRestClient() {
        super("/test.nav.no", mock(SystemUserTokenProvider.class));
    }

    @Override
    protected String utforRequest(Foedselsnummer fnr) {
        return FileToJson.toJson("/arbeidsforhold/arbeidsforhold.json");
    }
}

File: src/test/kotlin/no/nav/fo/veilarbregistrering/arbeidsforhold/adapter/StubAaregRestClient.kt
package no.nav.fo.veilarbregistrering.arbeidsforhold.adapter

import io.mockk.mockk
import no.nav.fo.veilarbregistrering.FileToJson.toJson
import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer
import okhttp3.Request

internal class StubAaregRestClient : AaregRestClient(mockk(relaxed = true), "/test.nav.no", mockk()) {
    override fun utfoer(request: Request) = toJson("/arbeidsforhold/arbeidsforhold.json")
    override fun buildRequest(fnr: Foedselsnummer): Request = buildRequest()

    private fun buildRequest() = Request.Builder().url("https://localhost:9090").build()
}